pyiron_base/jobs/flex/pythonfunctioncontainer.py

```diff
@@ -44,6 +44,8 @@ def __init__(self, project, job_name):
         super().__init__(project, job_name)
         self._function = None
         self._executor_type = None
+        self._mangle_name_on_save = True  # Automatically rename job using function and
```

Comment:
```suggestion
        self._overwrite_job_name_by_hash = True
```
 I feel when we have to explain the variable name in a comment, then maybe the variable name is not intuitive.

Reply:
Yeah, I was aware of this deficiency -- I fixed it in [the stacked PR](https://github.com/pyiron/pyiron_base/pull/1358/), but it was still unprofessional of me to just leave it knowingly-bad here.

I find your suggestion loses information on when/why we would rename this, and focuses on the implementation (hashing) instead of the functionality (leveraging input values), so I opted to go with `_automatically_rename_on_save_using_input`.

This is definitely an improvement, although in the stacked PR I'm still inclined to leave to comment to give even more depth to why we're doing this.